lib/glimesh_web/live/channel_settings_live/hosting.ex

```diff
@@ -0,0 +1,137 @@
+defmodule GlimeshWeb.ChannelSettingsLive.Hosting do
+  use GlimeshWeb, :live_view
+
+  alias Glimesh.ChannelHostsLookups
+  alias Glimesh.ChannelLookups
+  alias Glimesh.Streams.Channel
+  alias Glimesh.Streams.ChannelHosts
+
+  @impl true
+  def mount(_, session, socket) do
+    if session["locale"], do: Gettext.put_locale(session["locale"])
+
+    user = Glimesh.Accounts.get_user_by_session_token(session["user_token"])
+    channel = ChannelLookups.get_channel_for_user(user)
+    allow_hosting_changeset = Channel.change_allow_hosting(channel)
+    hosted_channels = ChannelHostsLookups.get_channel_hosting_list(channel.id)
+
+    {:ok,
+     socket
+     |> put_page_title(gettext("Hosting Settings"))
+     |> assign(:user, user)
+     |> assign(:channel, channel)
+     |> assign(:hosting_qualified, hosting_qualified?(user, channel))
+     |> assign(:allow_changeset, allow_hosting_changeset)
+     |> assign(:matches, [])
+     |> assign(:add_channel, "")
+     |> assign(:add_channel_selected_value, "")
+     |> assign(:hosted_channels, hosted_channels)}
+  end
+
+  defp hosting_qualified?(user, channel) do
+    account_age = Glimesh.Accounts.get_account_age_in_days(user)
+    total_streamed_hours = Glimesh.Streams.get_channel_hours(channel)
+
+    user.confirmed_at != nil and account_age >= 5 and total_streamed_hours >= 10
+  end
+
+  defp search_for_channels(user, term) do
+    ChannelLookups.search_hostable_channels_by_name(user, term)
+  end
+
+  @impl true
+  def handle_event("toggle_allow_hosting", %{"channel" => channel_params}, socket) do
+    case Channel.update_allow_hosting(socket.assigns.channel, channel_params) do
+      {:ok, channel} ->
+        {:noreply,
+         socket
+         |> assign(:channel, channel)
+         |> assign(:allow_changeset, Channel.change_allow_hosting(channel))
+         |> put_flash(:hosting_info, gettext("Saved Hosting Preference."))}
```

Comment:
I use "hosting_info" instead of just "info" for the flash messages due to a bug I've reported: [785](https://github.com/Glimesh/glimesh.tv/issues/785)